src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -0,0 +1,183 @@
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertSwitchStatementToExpression
+{
+    public partial class ConvertSwitchStatementToExpressionTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new ConvertSwitchStatementToExpressionDiagnosticAnalyzer(), new ConvertSwitchStatementToExpressionCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    int M(int i)
+    {
+        [||]switch (i)
+        {
+            case 1:
+                return 4;
+            case 2:
+                return 5;
+            case 3:
+                return 6;
+            default:
+                return 7;
+        }
+    }
+}",
+@"class Program
+{
+    int M(int i)
+    {
+        return i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6,
+            _ => 7
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestNested()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    int M(int i, int j)
+    {
+        [||]switch (i)
+        {
+            case 1:
+                switch (j)
+                {
+                    case 7:
+                        return 10;
+                    case 8:
+                        return 11;
+                    case 9:
+                        return 12;
+                }
+                break;
+            case 2:
+                return 5;
+            case 3:
+                return 6;
+            default:
+                throw null;
+        }
+    }
+}",
+@"class Program
+{
+    int M(int i, int j)
+    {
+        return i switch
+        {
+            1 => j switch
+            {
+                7 => 10,
+                8 => 11,
+                9 => 12
+            },
+            2 => 5,
+            3 => 6,
+            _ => throw null
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestSingleAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        int j;
+        [||]switch (i)
+        {
+            case 1:
+                j = 4;
+                break;
+            case 2:
+                j = 5;
+                break;
+            case 3:
+                j = 6;
+                break;
+        }
+    }
+}",
+@"class Program
+{
+    void M(int i)
+    {
+        int j;
+        j = i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6
```

Comment:
is the trailing comma optional or is it disallowed?  For symmetry purposes this looks really unclean to me (especially coming from 'go').  It also makes copy/pasting the last line to add more cases more unpleasant.  If this is optional, i would strongly consider us adding in that trailing comma.  If it's not allowed, i'd say it shoudl be allowed.  Tagging @gafter for this language design point.